### DISCUSSION TITLE
iOS build error code 301

### DISCUSSION THREAD

**Discussion by yvant** [airsdk/Adobe-Runtime-Support#2273] [General]
Hi, 

I'm getting a blank screen when testing my game on iOS with an AdHOC build. I'm using FlashDevelop on Windows,  and tried SDKs 33.1.1.929 and 33.1.1.935.

Here's a partial log I got from XCode:

00:15:18.379698+0100	general	AMSURLSession: [3B9DD385] Session decoder failed. Result = (null); Error = Error Domain=AMSErrorDomain Code=301 "Invalid Status Code" UserInfo={NSLocalizedFailureReason=The response has an invalid status code, AMSURL=https://sandbox.itunes.apple.com/commerce/messages/app/inbox?bundleId=xxxxx, AMSStatusCode=412, NSLocalizedDescription=Invalid Status Code}	com.apple.AppleMediaServices	AppleMediaServices	error	appstored

00:15:18.379863+0100	general	AMSURLSession: [3B9DD385] Task completed with error = Error Domain=AMSErrorDomain Code=301 "Invalid Status Code" UserInfo={NSLocalizedFailureReason=The response has an invalid status code, AMSURL=https://sandbox.itunes.apple.com/commerce/messages/app/inbox?bundleId=xxxxx, AMSStatusCode=412, NSLocalizedDescription=Invalid Status Code}	com.apple.AppleMediaServices	AppleMediaServices	error	appstored

00:15:18.379909+0100	StoreKit	Error in messages response: Error Domain=AMSErrorDomain Code=301 "Invalid Status Code" UserInfo={NSLocalizedFailureReason=The response has an invalid status code, AMSURL=https://sandbox.itunes.apple.com/commerce/messages/app/inbox?bundleId=xxxxx, AMSStatusCode=412, NSLocalizedDescription=Invalid Status Code}	com.apple.appstored	appstored	error	appstored

00:15:18.426465+0100	general	AMSURLProtocolHandler: [SKUT:S:L/xxxxx] Failed to add cookies for response. error = (null)	com.apple.AppleMediaServices	AppleMediaServices	error	appstored

I don't have any clue on how to solve it, I tried to regress a bit but it didn't solve the issue.
Any clue?

Thanks!

**Comment by idanasher** [comment]
I am having the exact same error.

any tips here ?

**Comment by idanasher** [comment]
I have opened an issue here:
https://github.com/airsdk/Adobe-Runtime-Support/issues/2387

**Comment by pwiesinger** [comment]
Any updates on this issue? We are currently facing the same problem...

**Reply by ajwfrost** [reply]
We'd missed seeing this issue.. just took a quick look and it seems related to StoreKit usage - are you using this (or an ANE that provides this functionality)? What's your actual behaviour, do you get a white screen (and if so, what's happening in something like Scout, do you see any error information or does it stop reporting frames, does it disconnect etc) - or does it crash (in which case can we get the IPS file)?

I would hope that any failure with the StoreKit APIs wouldn't have a knock-on effect to the application but of course I don't know what the application is doing or how it handles these kinds of things, it doesn't appear to be an issue within the core AIR functionality. Wondering whether the first step is to check the ANE behaviour in these cases?

thanks

**Reply by marchbold** [reply]
@pwiesinger Are you certain this log error is the cause of the issue? I have seen this log in certain conditions using StoreKit however the application isn't affected. Are you blocking your app awaiting InAppBilling requests? Certain things work differently in the sandbox

**Reply by pwiesinger** [reply]
Hello everyone. Sorry for the late response. We found the underlying issue and it turns out the log messages in the initial post had absolutely nothing to do with it. It was actually a firebase related issue... Sorry for the false alarm!
